### ISSUE TITLE
SSLv3 Alert Handshake Failure

### ISSUE THREAD

**Issue by ttaggorf** [MrAsterisco/homebridge-hisense-tv#52]
**Describe The Bug:**

When running the authorisation script, an error is produced which shows the below:

`root@homebridge-vm:/var/lib/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin $ python3.8 hisensetv.py 172.16.0.13 --authorize  --ifname eth0
[2024-03-28 19:27:12,472] [INFO    ] Unverified SSL context created.
[2024-03-28 19:27:12,473] [INFO    ] Network interface MAC Address: 00:15:5d:00:23:00
Traceback (most recent call last):
  File "hisensetv.py", line 6, in <module>
    __main__.main()
  File "/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__main__.py", line 128, in main
    with tv:
  File "/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 136, in __enter__
    self._mqtt_client.connect(self.hostname, self.port)
  File "/usr/local/lib/python3.8/dist-packages/paho/mqtt/client.py", line 914, in connect
    return self.reconnect()
  File "/usr/local/lib/python3.8/dist-packages/paho/mqtt/client.py", line 1073, in reconnect
    sock.do_handshake()
  File "/usr/lib/python3.8/ssl.py", line 1309, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLError: [SSL: SSLV3_ALERT_HANDSHAKE_FAILURE] sslv3 alert handshake failure (_ssl.c:1131)`

**To Reproduce:**
Run the script `python3.8 hisensetv.py 172.16.0.13 --authorize  --ifname eth0`

**Expected behavior:**
Should allow user to enter TV authorisation code.

**Plugin Config:**

```
    {
            "devices": [
                {
                    "id": "TV1",
                    "name": "Living Room TV",
                    "ipaddress": "172.16.0.13",
                    "macaddress": "38:64:07:ee:79:26",
                    "sslmode": "default"
                }
            ],
            "platform": "HiSenseTV"
    
```

**Screenshots:**

**Environment:**

* **Plugin Version**: 2.0.0
* **Homebridge Version**:  
* **Node.js Version**: 
* **NPM Version**: 
* **Operating System**:

**Comment by MrAsterisco** [comment]
Hi @ttaggorf !

Have you checked https://github.com/MrAsterisco/homebridge-hisense-tv/issues/28? It is the exact same error and it was solved by using the correct SSL certificates.

**Comment by ttaggorf** [comment]
Thanks for that. I have now resolved the above. 

However, now getting the below error instead...

```
sysadmin@pve-homebridge-1:/var/lib/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin$ python3 hisensetv.py 172.16.0.13 --authorize --ifname ens18

[2024-04-03 12:16:05,478] [INFO    ] Unverified SSL context created.
[2024-04-03 12:16:05,479] [INFO    ] Network interface MAC Address: bc:24:11:87:ec:e9

Traceback (most recent call last):
  File "/var/lib/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv.py", line 6, in <module>
    __main__.main()
  File "/var/lib/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__main__.py", line 128, in main
    with tv:
  File "/var/lib/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 124, in __enter__
    self._mqtt_client = mqtt.Client(self.client_id)
  File "/home/sysadmin/.local/lib/python3.10/site-packages/paho/mqtt/client.py", line 766, in __init__
    raise ValueError(

ValueError: Unsupported callback API version: version 2.0 added a callback_api_version, see migrations.md for details
```

Google isn't finding my results unfortunately.

**Comment by ttaggorf** [comment]
> Hi @ttaggorf !
> 
> Have you checked #28? It is the exact same error and it was solved by using the correct SSL certificates.

Thanks. All sorted on that front, albeit now getting unsupported callback issues.

```
sysadmin@pve-homebridge-1:/var/lib/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin$ python3.8 hisensetv.py 172.16.0.13 --authorize --ifname ens18 --certfile tvcert.cer --keyfile tvkey.key

[2024-04-03 13:04:31,524] [INFO    ] SSL context created with cert file (tvcert.cer) and private key (tvkey.key)

[2024-04-03 13:04:31,524] [INFO    ] Network interface MAC Address: bc:24:11:87:ec:e9

Traceback (most recent call last):
  File "hisensetv.py", line 6, in <module>
    __main__.main()
  File "/var/lib/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__main__.py", line 130, in main
    with tv:
  File "/var/lib/homebridge/node_modules/homebridge-hisense-tv-remotenow/bin/hisensetv/__init__.py", line 124, in __enter__
    self._mqtt_client = mqtt.Client(self.client_id)
  File "/usr/local/lib/python3.8/dist-packages/paho/mqtt/client.py", line 766, in __init__
    raise ValueError(

ValueError: Unsupported callback API version: version 2.0 added a callback_api_version, see migrations.md for details
```

**Comment by MrAsterisco** [comment]
Great to hear this was fixed! It seems you're using an incorrect version of Python, though: can you please check [the Requirements](https://github.com/MrAsterisco/homebridge-hisense-tv?tab=readme-ov-file#requirements) section?

**Comment by LeLunZ** [comment]
Closed because main issue was solved and new issues #53 was opened.
